homeassistant/components/velux/light.py

```diff
@@ -0,0 +1,85 @@
+"""Support for Velux lights."""
+from pyvlx import Intensity, LighteningDevice
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    SUPPORT_BRIGHTNESS,
+    LightEntity,
+)
+from homeassistant.core import callback
+
+from . import DATA_VELUX
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up light(s) for Velux platform."""
+    entities = []
+    for node in hass.data[DATA_VELUX].pyvlx.nodes:
+        if isinstance(node, LighteningDevice):
+            entities.append(VeluxLight(node))
+    async_add_entities(entities)
+
+
+class VeluxLight(LightEntity):
+    """Representation of a Velux light."""
+
+    def __init__(self, node):
```

Comment:
Would it make sense to create a `VeluxEntity` to abstract some of the velux logic, which is used in all platforms?

Reply:
It's only 3 methods which are similar, which don't contain much logic. I'd like to have DRY code, but in this case it seems an unnecessary abstraction/complexity to me.

Reply:
Would maybe good in a follow up PR. The repetition would be in `__init__`, `async_register_callbacks`, `async_added_to_hass`, `unique_id`, `name`? And for example `device_info` could be added, to enrich the entity registry. 

See Somfy integration for an example;
https://github.com/home-assistant/core/blob/a59460a23336627d0bc12b1eefffdaa516e55e87/homeassistant/components/somfy/__init__.py#L182

Reply:
I have added `VeluxEntity` abstraction which `cover` and `light` entities reuse.
The device info would be good for a follow up PR.